modules/auxiliary/admin/dcerpc/cve_2020_1472_zerologon.rb

```diff
@@ -0,0 +1,195 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+
+  include Msf::Exploit::Remote::DCERPC
+  include Msf::Exploit::Remote::SMB::Client
+  include Msf::Auxiliary::Report
+
+  CheckCode = Exploit::CheckCode
+  Netlogon = RubySMB::Dcerpc::Netlogon
+  EMPTY_SHARED_SECRET = OpenSSL::Digest::MD4.digest('')
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Netlogon Weak Cryptographic Authentication',
+        'Description' => %q{
+          A vulnerability exists within the Netlogon authentication process where the security properties granted by AES
+          are lost due to an implementation flaw related to the use of a static initialization vector (IV). An attacker
+          can leverage this flaw to target an Active Directory Domain Controller and make repeated authentication attempts
+          using NULL data fields which will succeed every 1 in 256 tries (~0.4%). This module leverages the vulnerability
+          to reset the machine account password to an empty value, which will then allow the attacker to authenticate as
+          the machine account. After exploitation, it's important to restore this password to it's original value. Failure
+          to do so can result in service instability.
+        },
+        'Author' => [
+          'Tom Tervoort', # original vulnerability details
+          'Spencer McIntyre', # metasploit module
+          'Dirk-jan Mollema' # password restoration technique
+        ],
+        'Notes' => {
+          'AKA' => [ 'Zerologon' ]
+        },
+        'License' => MSF_LICENSE,
+        'Actions' => [
+          [ 'REMOVE', { 'Description' => 'Remove the machine account password' } ],
+          [ 'RESTORE', { 'Description' => 'Restore the machine account password' } ]
+        ],
+        'DefaultAction' => 'REMOVE',
+        'References' => [
+          [ 'CVE', '2020-1472' ],
+          [ 'URL', 'https://www.secura.com/blog/zero-logon' ],
+          [ 'URL', 'https://github.com/SecuraBV/CVE-2020-1472/blob/master/zerologon_tester.py' ],
+          [ 'URL', 'https://github.com/dirkjanm/CVE-2020-1472/blob/master/restorepassword.py' ]
+        ]
+      )
+    )
+
+    register_options(
+      [
+        Opt::RPORT(0),
```

Comment:
Since `0` means automatic RPC port discovery, you may want to note that in the option's description.